src/parser/hunter/shared/modules/talents/AMurderOfCrows.js

```diff
@@ -43,29 +40,32 @@ class AMurderOfCrows extends Analyzer {
       return;
     }
     this.casts++;
-    this.enemyID = encodeTargetString(event.targetID, event.targetInstance);
-    this.applicationTimestamp = event.timestamp;
+    this.applicationTimestamp = null;
     this.lastDamageTick = null;
   }
 
   on_byPlayer_damage(event) {
     const spellId = event.ability.guid;
-    if (this.lastDamageTick && event.timestamp + MS_BUFFER < this.applicationTimestamp + CROWS_DURATION && event.timestamp > this.lastDamageTick + CROWS_TICK_RATE + MS_BUFFER && this.spellUsable.isOnCooldown(SPELLS.A_MURDER_OF_CROWS_TALENT.id)) {
+    /**
+     * Checks if we've had atleast 1 damage tick of the currently applied crows, and checks that crows is in fact on cooldown.
+     * Then it checks whether the current damage event is less than the full duration of crows, followed by a check to see if more than 1 second has passed since last tick
+     * If more than 1 second has passed, we can assume that crows has been reset, and thus we the CD.
+     */
+    if (this.lastDamageTick && this.spellUsable.isOnCooldown(SPELLS.A_MURDER_OF_CROWS_TALENT.id) && event.timestamp + MS_BUFFER < this.applicationTimestamp + CROWS_DURATION + MS_BUFFER && event.timestamp > this.lastDamageTick + CROWS_TICK_RATE + MS_BUFFER) {
       this.spellUsable.endCooldown(SPELLS.A_MURDER_OF_CROWS_TALENT.id, event.timestamp);
-      this.lastDamageTick = null;
-      this.enemyID = null;
     }
     if (spellId !== SPELLS.A_MURDER_OF_CROWS_DEBUFF.id) {
       return;
     }
-    if (!this.enemyID) {
-      this.enemyID = encodeTargetString(event.targetID, event.targetInstance);
-    }
     if (this.casts === 0) {
       this.casts++;
       this.spellUsable.beginCooldown(SPELLS.A_MURDER_OF_CROWS_TALENT.id, this.owner.fight.start_time);
       this.applicationTimestamp = this.owner.fight.start_time;
     }
+    //This accounts for the travel time of crows, since the first damage marks the time where the crows debuff is applied
```

Comment:
Why not use applydebuff instead of damage event?

Reply:
There is no applydebuff event anymore, which is what broke the original implementation in the first place

Reply:
oh lul, that's a weird one